src/panels/config/automation/condition/types/ha-automation-condition-time.ts

```diff
@@ -36,10 +62,80 @@ export class HaTimeCondition extends LitElement implements ConditionElement {
         .value=${before}
         @value-changed=${this._valueChanged}
       ></paper-input>
+
+      ${this._days.map(
+        (day) => html`
+          <ha-formfield
+            class="flex"
+            .label=${this.hass!.localize(
+              `ui.panel.config.automation.editor.conditions.type.time.weekdays.${day.name}`
+            )}
+            .dir=${computeRTLDirection(this.hass!)}
+            alignEnd
+            spaceBetween
+          >
+            <ha-switch
+              .day=${day.name}
+              .checked=${weekday ? weekday.includes(day.name) : false}
+              @change=${this._dayValueChanged}
+            >
+            </ha-switch>
+          </ha-formfield>
+        `
+      )}
     `;
   }
 
   private _valueChanged(ev: CustomEvent): void {
     handleChangeEvent(this, ev);
   }
+
+  private _dayValueChanged(ev: CustomEvent): void {
+    ev.stopPropagation();
+
+    const day = (ev.currentTarget as any).day;
+    const checked = (ev.currentTarget as HaSwitch).checked;
```

Comment:
```ts
interface WeekdayHaSwitch extends HaSwitch {
	day: string;
}
```

```suggestion
	const daySwitch = (ev.currentTarget as WeekdayHaSwitch);
```

Then use `daySwitch.day` and `daySwitch.checked`

Reply:
I did nott think about that. Fixed